client/src/data.js

```diff
@@ -1,37 +1,47 @@
-let data = {
-  entries: [],
-  nextEntryId: 1,
-};
-
-window.addEventListener('beforeunload', function (event) {
-  const dataJSON = JSON.stringify(data);
-  localStorage.setItem('code-journal-data', dataJSON);
-});
-
-const localData = JSON.parse(localStorage.getItem('code-journal-data'));
-if (localData) {
-  data = localData;
-}
-
-export function readEntries() {
-  return data.entries;
-}
-
-export function addEntry(entry) {
-  entry.entryId = data.nextEntryId++;
-  data.entries.unshift(entry);
+export async function addEntry(entry) {
+  try {
+    const reqMethod = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(entry),
+    };
+    const response = await fetch('/api/entries', reqMethod);
+    if (!response.ok) {
+      throw new Error(`fetching error status ${response.status}`);
+    }
+  } catch (err) {
+    console.error(err);
```

Comment:
If you catch the error and log it here, the code continues executing. The next thing that happens in this function is you reach the end, which will instruct JavaScript to return `undefined`.

Therefore, you have logged the error but you have not given the user any feedback on what happened (users don't usually look at console logs), and you have not give the code that called you a chance to do anything with the error.

The rule with `await` is that it must be wrapped in a try/catch. However, that try/catch can appear anywhere in the call stack. The best way to handle these types of functions that are just doing backend work is to _not_ use try/catch here in the function but to rely on the caller using try/catch. That way, the exception will bubble up from this function to the caller and be caught there.

This applies to all the functions in this file.